scripts/ci-entrypoint.sh

```diff
@@ -139,6 +139,10 @@ install_addons() {
     # export the target cluster KUBECONFIG if not already set
     export KUBECONFIG="${KUBECONFIG:-${PWD}/kubeconfig}"
 
+    # wait for the apiserver pod to be Ready.
+    APISERVER_POD=$(${KUBECTL} get pods -n kube-system -o name | grep apiserver)
+    "${KUBECTL}" wait --for=condition=Ready -n kube-system "${APISERVER_POD}" --timeout=5m
```

Comment:
Are we confident that this wait for a single observation of `Ready` is sufficient to deal w/ observed apiserver weirdness after updating static pods? More specifically, are we confident that this wait check will trigger *after* the control plane static pod spec files have been modified?

Reply:
Yes, the apiserver pod stays pending until it's ready and then it's ready for good

That's why I changed it to this instead of retrying actually because it's the best signal of "things are stable and ready for good now" from what I've observed. If we add retries instead we'd have to add retries to every single place that talks to the apiserver because there is no way of controlling the timing of when the apiserver will become temporarily unavailable before that pod `Ready` transition happens.

Reply:
I can try that locally and see what happens

Reply:
yeah that's equivalent, pushed the changes